### PR TITLE
Multicloud blog post

### PR DESCRIPTION
**Pull request by dogle-scottlogic**
Please add a direct link to your post here:

[https://<your-username>.github.io/blog/<post-url>](https://dogle-scottlogic.github.io/blog/2024/10/18/multicloud.html)

Have you (please tick each box to show completion):

 - [x] Added your blog post to a single category?
 - [x] Added a brief summary for your post? Summaries should be roughly two sentences in length and give potential readers a good idea of the contents of your post.
 - [x] Checked that the build passes?
 - [x] Checked your spelling (you can use `npm install` followed by `npx mdspell "**/{FILE_NAME}.md" --en-gb -a -n -x -t` if that's your thing)
 - [x] Ensured that your author profile contains a profile image, and a brief description of yourself? (make it more interesting than just your job title!)
 - [x] Optimised any images in your post? They should be less than 100KBytes as a general guide.

Posts are reviewed / approved by your Regional Tech Lead.

### CODE REVIEW

File: _posts/2024-10-18-multicloud.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Vendor Lock-in
+
+<img src="{{site.baseurl}}/dogle/assets/multicloud/li.jpg" alt="Lock in pub sign" title="You can check out any time you like, but you can never leave" style="display: block; margin: 1rem auto;" />
+
+Vendor lock in is fast becoming a real issue. Over the last few years the three big players in Cloud Computing (AWS, GCP, Azure) have dominated the market with AWS especially taking the lead. The problem is of course, once you're in, it's very hard to get out again. Cloud providers are falling over themselves to help you move into the cloud via all sorts of solutions from software to literally driving a truck up to your office full of hard drives.
```
Comment (chriswilty) [inline _posts/2024-10-18-multicloud.md]: "Lock-in" or "lock in"?

File: _posts/2024-10-18-multicloud.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Today most software projects utilise some form of cloud computing to host their infrastructure, often choosing one of the big three providers, Google (GCP), Microsoft (Azure) or Amazon (AWS). What I think is less common, is to consider multiple cloud providers when looking to host resources although it is something which is mentioned from time to time. For example, Monzo has taken steps to address resiliency though running a [live alternative system](https://www.linkedin.com/posts/joemerriman1989_monzo-banks-multi-cloud-environment-is-now-activity-7196510667674132481-7o5y/) in an alternate cloud provider. DataStax provides services to help deploy the Cassandra database [across multi-cloud](https://www.datastax.com/blog/datastax-helps-enterprises-achieve-multi-cloud-deployments-open-source-apache) and MongoDB's Atlas also [facilitates multi-cloud deployment](https://www.mongodb.com/resources/basics/multicloud). Solutions and implementations are out there then, but what is the problem that we're solving here and why would we bother?
```
Comment (chriswilty) [inline _posts/2024-10-18-multicloud.md]: "across multi-cloud" is not very descriptive link text, for SEO purposes; I suggest including "Cassandra database" as well.

Equally, I'd use "MongoDB's Atlas also facilitates multi-cloud" as the link text.

File: _posts/2024-10-18-multicloud.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+<img src="{{site.baseurl}}/dogle/assets/multicloud/aws-snowmobile.jpg" alt="AWS Snowmobile" title="AWS Snowmobile" style="display: block; margin: 1rem auto;" />
+
+What they are less keen on, unsurprisingly, is helping you move your resources out of their particular cloud. As governments have moved public services into the cloud as part of digital strategies this has already been [called out](https://www.theregister.com/2024/04/04/uk_cddo_admits_cloud_spending_lock_issues_exclusive/) as a big concern. It's costly and difficult to move out of a cloud provider once you're there. To be clear this isn't an accident or unavoidable necessity of working in the cloud, cloud providers make efforts to ensure that their home-grown services are easier to use and provide a greater range of built in functionality than other services they offer and they make sure to promote these services as much as possible. As an example, when studying for an AWS exam recently I couldn't help but notice that many of the questions in the exam were tailored to promote AWS native services or around how you might move your existing infrastructure into AWS easily.
```
Comment (chriswilty) [inline _posts/2024-10-18-multicloud.md]: Similarly, "called out" might be better as "called out as a big concern", though that's arguably equally vague. Not sure what to do with this one...

File: _posts/2024-10-18-multicloud.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+You could make a concerted effort from the start to not let yourself get too dependant on a single cloud providers offering, steering clear of SaaS offerings native to a given cloud provider and trying to use more generic resources common to all providers. The problem is that even these common resource are not truly generic. An Amazon EC2 instance is not configured in the same way as an Azure Virtual Machine (AVM), even if the end result is similar. This means that while technically, if you had to, you could recreate everything in a different provider, the cost of doing so in real terms would be huge. It would be a massive undertaking as you can't just drag and drop existing architecture into a different cloud, but would need to rebuild from scratch each resource. It would be much too costly to be a consideration in anything but the most extreme case.
```
Comment (chriswilty) [inline _posts/2024-10-18-multicloud.md]: > cloud provider's offering

Missing apostrophe.

File: _posts/2024-10-18-multicloud.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+A step further again would be to have a for cloud providers to adhere to a common API as a standard, removing the need to transform our IAC to the correct structure for the given provider upfront. Efforts such as the [Finos Common Cloud Controls Project](https://www.finos.org/common-cloud-controls-project) are moving in this direction while specifically looking at compliance issues for the finance sector, but in my opinion, until cloud providers are forced to conform to a standard API for deployment of common resources they will continue to be able to hold all those using their services, from governments to banks, over a barrel.
```
Comment (chriswilty) [inline _posts/2024-10-18-multicloud.md]: > would be *to have a* for cloud providers to adhere to a common API
> would be to have a *for cloud providers to adhere to a* common API

I think you've probably changed your mind but left both choices in here!

Comment (chriswilty) [inline _posts/2024-10-18-multicloud.md]: You might consider linking to the recent blog post on CCC, for a bit of self-promotion 😉

**Review by chriswilty — COMMENTED** [review]
Looks good to me, love the image titles! Just a couple of typos and suggestions on link texts.

File: _posts/2024-10-18-multicloud.md (hunk 1 above)
Comment (dogle-scottlogic) [reply _posts/2024-10-18-multicloud.md]: haha I couldn't decide - also multicloud vs multi-cloud tripped me up - I'll just pick one already 😆

File: _posts/2024-10-18-multicloud.md (hunk 5 above)
Comment (dogle-scottlogic) [reply _posts/2024-10-18-multicloud.md]: ah I did consider that but didn't think it was published yet - I'll change the link 👍

**Review by chriswilty — APPROVED** [review]
Lovely!

**Pull request merged**